### PR TITLE
UI: Bandwidth test & "No stream key" dialogs to Stream Settings button

### PR DESCRIPTION
**Pull request by WizardCM**
### Description

In #2203 @JohannMG added a dialog with a button to open Settings. In the original PR, they mentioned wanting the button to go direct to the Stream page of the Settings window. This adds that capability to the button, alongside the ability to open any Settings screen.

Additionally, it adds a similar "Open Settings" button to the Bandwidth Test notification, providing users a way to know where to go to disable the setting.

image

### Motivation and Context

Commit 1 accomplishes what @JohannMG originally intended to do, with additional code for future-proofing.

Commit 2 adds a Settings button to Bandwidth Test because we've had a number of users ask in Support "How do I turn it off" because the dialog doesn't provide much context. This provides a way to go in and easily turn off Bandwidth Test mode.

### How Has This Been Tested?

Try streaming without a stream key configured. Click "Open Settings" and see that the Stream tab has been pre-selected.

Try streaming with Bandwidth Test mode enabled (tested with Twitch Service Integration). Click "Open Settings" and see that the Stream tab has been pre-selected.

### Types of changes
- New feature (non-breaking change which adds functionality)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by Fenrirthviti** [comment]
I like the idea, but I feel the dialog message is a bit too wordy. We should keep things as minimal and simple to read as possible, as the more text you have the less likely users are to read it. Maybe something like:

"You have OBS configured for bandwidth test mode. This can be disabled in your stream settings. Do you want to continue?"

File: UI/window-basic-main.cpp [hunk 1]
```diff
@@ -6081,8 +6088,8 @@ void OBSBasic::on_streamButton_clicked()
 		switch (action) {
 		case StreamSettingsAction::ContinueStream:
 			break;
-		case StreamSettingsAction::OpenSettings:
-			on_action_Settings_triggered();
+		case StreamSettingsAction::OpenSettingsStream:
+			on_action_Settings_triggered(1);
```
Comment (JohannMG) [inline UI/window-basic-main.cpp:6092]: I think this would be a great place for an enum like `ObsSettingsOpen::StreamSettings` in case the menu is rearranged later, intent would be more clear.

File: UI/ui-validation.cpp [hunk 2]
```diff
@@ -117,3 +117,47 @@ UIValidation::StreamSettingsConfirmation(QWidget *parent, OBSService service)
...
+		QPushButton *settings;
+		QPushButton *stream;
+
+#ifdef __APPLE__
+#define ACCEPT_BUTTON QMessageBox::AcceptRole
+#define REJECT_BUTTON QMessageBox::ResetRole
+#else
+#define ACCEPT_BUTTON QMessageBox::NoRole
+#define REJECT_BUTTON QMessageBox::NoRole
+#endif
```
Comment (JohannMG) [inline UI/ui-validation.cpp:144]: Is this needed?  Looks like copy paste from above code that already defines these. 
Either remove or consolidate&foist these to the top of the file.

File: UI/window-basic-main.cpp [hunk 3]
```diff
@@ -6100,21 +6100,21 @@ void OBSBasic::on_streamButton_clicked()
...
-				OBSMessageBox::question(
-					this, QTStr("ConfirmBWTest.Title"),
-					QTStr("ConfirmBWTest.Text"));
+		auto bwAction =
+			UIValidation::BandwidthModeConfirmation(this, service);
+		switch (bwAction) {
+		case StreamSettingsAction::ContinueStream:
+			break;
+		case StreamSettingsAction::OpenSettingsStream:
+			on_action_Settings_triggered(1);
```
Comment (JohannMG) [inline UI/window-basic-main.cpp:6109]: oh I see. Further on enum usage, I think take this parsing out of Main Window and just send settings the enum to figure itself out?

**Comment by Fenrirthviti** [comment]
I disagree on that point about it being complex. It's extremely simple, and complicating the dialog by explaining what bandwidth mode is seems like a waste of text. Just say you're in bandwidth mode, explain where it can be turned off, ask to continue. If we want to put a lengthy explanation of what bandwidth mode is, it can be in the tooltip on the option itself.

**Comment by JohannMG** [comment]
> I disagree on that point about it being complex.

Half the wizard's workflow is focused on bandwidth testing: I'd say users general have a weak understanding what it even means for video. Tooltips aren't as discoverable and inconsistent. If turning the mode off is a blocker for streaming keeping education is important .

Or we could change the dialog to `Settings, Continue, and Disable Bandwidth Test Mode` so they do not have to jump into settings at all.

**Comment by PatTheMav** [comment]
In the interest of either getting this merged or closed (it's been sitting on the pile for 5 years now), can someone give me a quick rundown of what this changes and how I can test/observe this locally? I never used the bandwidth test stuff, so I have no idea how any of this works and what the user need is that this solves.

**Comment by Fenrirthviti** [comment]
You can test just by setting Twitch as the output, and leaving stream key blank and hitting "Start Streaming", or similarly by checking the "Enable Bandwidth Test Mode" in the Stream options for Twitch.

For reference, this is what the current dialog shows for blank key:
image

and for bandwidth test mode:
image

This change is allowing the "Open Settings" button to jump right to the stream page, and adds the same button to the bandwidth test dialog.

The wording changes I previously requested are out of scope, I didn't realize at the time they were already merged in a separate PR, so they can be addressed separately (I'm still not the biggest fan of the current verbiage).

**Comment by Warchamp7** [comment]
Closing old PRs that require a rebase for the big frontend reorganization we did. Feel free to rebase and re-open a new pull request for this.

Please note this does not necessarily mean the change itself will be accepted. Review and consideration will happen if it's re-opened.
